src/RewardsDistribution.t.sol

```diff
@@ -0,0 +1,253 @@
+import {DssVestWithGemLike} from "./interfaces/DssVestWithGemLike.sol";
+import {StakingRewardsLike} from "./interfaces/StakingRewardsLike.sol";
+import {SDAO} from "./SDAO.sol";
+import {RewardsDistribution} from "./RewardsDistribution.sol";
+import {DistributionCalc, LinearIncreasingDistribution, ConstantDistribution} from "./DistributionCalc.sol";
+
+contract RewardsDistributionTest is DssTest {
+    using stdStorage for StdStorage;
+
+    RewardsDistribution dist;
+    DistributionCalc calc;
+    DssVestWithGemLike vest;
+    StakingRewardsLike farm;
+    SDAO token;
+
+    uint256 totalRewards = 1 * 10 ** 18;
+    uint256 duration = 360 days;
+
+    uint256 vestId;
+
+    function setUp() public {
+        // DssVest checks if params are not too far away in the future or in the past relative to `block.timestamp`.
+        // It has a 20 years interval check hardcoded, so we need to be at a time that is at least 20 years ahead of
+        // the Unix epoch.  We are setting the current date of the chain to 2000-01-01 to comply with that requirement.
+        vm.warp(946692000);
+
+        token = new SDAO("Token", "TKN");
+
+        vest = DssVestWithGemLike(deployCode("DssVest.sol:DssVestMintable", abi.encode(address(token))));
+        vest.file("cap", type(uint256).max);
+        farm = new MockStakingRewards(address(token), 7 days);
+        calc = new ConstantDistribution();
+        dist = new RewardsDistribution(address(vest), address(farm), address(calc));
+
+        vestId = _createVest();
+
+        dist.file("vestId", vestId);
+        // Allow DssVest to mint tokens
+        token.rely(address(vest));
+    }
+
+    function testDistribute() public {
+        // 1st distribution
+        skip(duration / 3);
+
+        assertEq(token.balanceOf(address(farm)), 0);
+
+        vm.expectEmit(false, false, false, true, address(farm));
+        emit RewardAdded(totalRewards / 3);
+        dist.distribute();
+
+        assertEq(token.balanceOf(address(farm)), totalRewards / 3);
+
+        // 2nd distribution
+        skip(duration / 3);
+
+        vm.expectEmit(false, false, false, true, address(farm));
+        emit RewardAdded(totalRewards / 3);
+        dist.distribute();
+
+        // Allow for 0,01% error tolerance due to rounding errors.
+        uint256 tolerance = 0.01e18;
+
+        assertApproxEqRel(token.balanceOf(address(farm)), (2 * totalRewards) / 3, tolerance);
+
+        // 3rd distribution
+        skip(duration / 3);
+
+        vm.expectEmit(false, false, false, true, address(farm));
+        emit RewardAdded(totalRewards / 3);
+        dist.distribute();
+
+        assertApproxEqRel(token.balanceOf(address(farm)), totalRewards, tolerance);
+    }
+
+    function testRegressionDistributeFromMultipleVests() public {
+        // 1st vest
+        skip(duration);
+
+        assertEq(token.balanceOf(address(farm)), 0);
+
+        vm.expectEmit(false, false, false, true, address(farm));
+        emit RewardAdded(totalRewards);
```

Comment:
Maybe one of these event checks should be the `Distribute` event.

Reply:
Yeah, `Distribute` makes more sense here.
Fixed.

Reply:
Fixed. Can't resolve comments.